demos/CMakeLists.txt

```diff
@@ -39,6 +39,7 @@ include_directories(
 
 set(DEMOS_SCRIPTS
     task_space_control_loop
+    kinematics_control_loop
 )
 
 foreach(SCRIPT ${DEMOS_SCRIPTS})
```

Comment:
I'm not sure it's the right approach to put that into a `for` loop and do it for each file because some files do not need to link all the libraries or compile definitions. On the other hand, I don't know how this actually affects the size or the overhead of the executable

Reply:
This is not a huge problem, since the code from the static libraries are only linked when they are referenced by the main source code. If you link a library that you don't use, the final executable will be unchanged as far as I am aware. Perhaps with dynamic libraries, you'd have a few more bytes in the header of the executable to store the path, but overall it's 

Really the main reason you don't want to link / include all the paths by default is to avoid namespace conflicts. Overall, in this case I don't think it's an issue. The for loop is a neat solution when the targets are all similar enough!

Reply:
Yes I thought about that. We can also remove the `target_link_library` macro for the for loop but then it would not save much compared to rewrite everything for each executables.